nexus/reconfigurator/planning/src/blueprint_builder.rs

```diff
@@ -305,6 +362,7 @@ impl<'a> BlueprintBuilder<'a> {
             sled_ip_allocators: BTreeMap::new(),
             zones: BlueprintZones::new(parent_blueprint),
             zones_in_service: parent_blueprint.zones_in_service.clone(),
+            expunged_nexus_zones: parent_blueprint.expunged_nexus_zones.clone(),
```

Comment:
Happy to discuss this during the meeting later today, but just jotting down a couple of thoughts here:

* Once a zone gets added to `expunged_nexus_zones`, how does the planner know when it's okay to remove it from a subsequent child blueprint?
* Does this need to be _Nexus_ zones specifically? Or could it be `zones_expunged` like `zones_in_service`? (I haven't thought about this much, but am assuming we're eventually going to want to track other kinds of zones that need execution work to be expunged.)

Reply:
> Happy to discuss this during the meeting later today, but just jotting down a couple of thoughts here:
> 
> * Once a zone gets added to `expunged_nexus_zones`, how does the planner know when it's okay to remove it from a subsequent child blueprint?

I was wondering the same thing. Should there also be a 'decommissioned' column for when the zone has been completely removed?

> * Does this need to be _Nexus_ zones specifically? Or could it be `zones_expunged` like `zones_in_service`? (I haven't thought about this much, but am assuming we're eventually going to want to track other kinds of zones that need execution work to be expunged.)

I was also thinking the same thing. It seems somewhat silly to have a separate table per zone type unless each of those tables would have different rows because they store other "checkpoint" information while decommissioning a zone. I don't think we were planning to do this inside the zone table, but I'm not sure.

Reply:
The latter we've taken into account by switching to a disposition enum.